gdal/gcore/gdal_priv.h

```diff
@@ -488,14 +488,18 @@ class CPL_DLL GDALDataset : public GDALMajorObject
 
     void ReportError(CPLErr eErrClass, CPLErrorNum err_no, const char *fmt, ...)  CPL_PRINT_FUNC_FORMAT (4, 5);
 
+    virtual char ** GetMetadata(const char * pszDomain = "");
+
+    virtual char ** GetMetadataDomainList();
+    
 private:
     void           *m_hPrivateData;
 
     OGRLayer*       BuildLayerFromSelectInfo(swq_select* psSelectInfo,
                                              OGRGeometry *poSpatialFilter,
                                              const char *pszDialect,
                                              swq_select_parse_options* poSelectParseOptions);
-
+    CPLStringList * papoDerivedMetadataList;
```

Comment:
And the naming convention for a CPLStringList* would be a poDerivedMetadataList (pointer to an object), or oDerivedMetadataList for a CPLStringList